packages/elements/src/SafeAreaProviderCompat.tsx

```diff
@@ -23,19 +24,23 @@ const initialMetrics =
       }
     : initialWindowMetrics;
 
-export default function SafeAreaProviderCompat({ children }: Props) {
+export default function SafeAreaProviderCompat({ children, style }: Props) {
   return (
     <SafeAreaInsetsContext.Consumer>
       {(insets) => {
         if (insets) {
           // If we already have insets, don't wrap the stack in another safe area provider
           // This avoids an issue with updates at the cost of potentially incorrect values
           // https://github.com/react-navigation/react-navigation/issues/174
-          return children;
+          return style == null ? (
+            children
+          ) : (
+            <View style={style}>{children}</View>
```

Comment:
This `View` needs to have `flex: 1`, otherwise the behavior will be different based on whether there's a parent `SafeAreaProvider` or not.

Reply:
Done.